ui/src/app/pages/data/data-set/genomic-extraction-modal.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+
+import {Button} from 'app/components/buttons';
+import {ErrorMessage} from 'app/components/messages';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {dataSetApi} from 'app/services/swagger-fetch-clients';
+
+import {DataSet} from 'generated/fetch';
+
+const {useState} = React;
+
+
+interface Props {
+  dataSet: DataSet;
+  workspaceNamespace: string;
+  workspaceFirecloudName: string;
+  closeFunction: Function;
+}
+
+export const GenomicExtractionModal = ({
```

Comment:
nit: pass in Workspace object instead of namespace/firecloud pair? They're inseparable concepts IMO so it's nicer to pass them around as a single object where possible.

Reply:
In this case, the workspace object is not the origin for this data. Rather, it's the `WorkspaceResource` (which has these fields). The reason not to use `WorkspaceREsource` directly is that this modal doesn't need to care about the `WorkspaceResource` model, e.g. if we want to reuse some of this code on the dataset builder page.